canals/component/component.py

```diff
@@ -181,9 +179,7 @@ def run(self, **kwargs):
                 return {"output_1": kwargs["value_1"], "output_2": ""}
         ```
         """
-        instance.__canals_input__ = {
-            name: InputSocket(name=name, type=type_, is_optional=_is_optional(type_)) for name, type_ in types.items()
```

Comment:
Do you think we need the `_is_optional` function at all in the library? We might as well delete it if it's not used anywhere else